### PR TITLE
SFR-1639/setUp_restApi_test

### PR DESCRIPTION
**Pull request by ayan1229**
This pr does the following

- intoduces restApi testing in QA ennvironment
- validates response for GET request for a collection
- validates the collection name and titles within the collection

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # CHANGELOG
```
Comment (kylevillegas93) [inline CHANGELOG.md:1]: Were these intended?

File: tests/restApi/test_get_request.py
Comment (kylevillegas93) [inline tests/restApi/test_get_request.py:1]: let's use snake case throughout naming. 

Is this file name explicit enough?

File: CHANGELOG.md (hunk 1 above)
Comment (ayan1229) [reply CHANGELOG.md:1]: I only added the following line.
`- implement restApi testing using pytest`

File: tests/restApi/test_get_request.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import requests
+import pytest
+
+
+BASE_URL = "https://drb-api-qa.nypl.org"
+url = BASE_URL + "/collection/3650664c-c8be-4d07-8d64-2d7003b02048"
+
+def test_get_request():
+    response = requests.get(url)
+    assert response.status_code == 200, f"Expected status code 200, but got {response.status_code}"
```
Comment (kylevillegas93) [inline tests/restApi/test_get_request.py:10]: For readability, I recommend using the following format. By separating the chunks with new lines, it's easier to parse.

```
test setup

api call under test

assertions/expectations
```

File: tests/restApi/test_get_request.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+def test_collection_name():
+
+    response = requests.get(url)
+    assert response.status_code == 200, f"Expected status code 200, but got {response.status_code}"
+
+    # Parse the JSON response
```
Comment (kylevillegas93) [inline tests/restApi/test_get_request.py:17]: are these comments necessary?

File: tests/restApi/test_get_request.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+def test_validate_json_fields():
+   
+    response = requests.get(url)
+    assert response.status_code == 200, f"Expected status code 200, but got {response.status_code}"
+
+    # Parse the JSON response
+    data = response.json()
+
+    # Validate "numberOfItems" is 2
+    assert data.get('metadata', {}).get('numberOfItems') == 2, f"Expected 'numberOfItems' to be 2, but got {data.get('metadata', {}).get('numberOfItems')}"
```
Comment (kylevillegas93) [inline tests/restApi/test_get_request.py:37]: What happens if the `numberOfItems` changes?

File: tests/restApi/test_get_request.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    # Validate the first title is "New Mexico magazine "
+    first_publication_title = data.get('publications', [])[0].get('metadata', {}).get('title')
```
Comment (kylevillegas93) [inline tests/restApi/test_get_request.py:40]: What happens if the first publication changes?

File: tests/restApi/test_get_request.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    # Validate the title in the metadata
+    expected_title = "Works by Ayan"
+    actual_title = data.get("metadata", {}).get("title", "")
+    
+    # Assert that the title is "Works by Ayan"
+    assert actual_title == expected_title, f"Expected title '{expected_title}', but got '{actual_title}'"
+
+
+def test_validate_json_fields():
```
Comment (kylevillegas93) [inline tests/restApi/test_get_request.py:28]: Could we just simply have 1 test with our expectations for a GET request that returns successfully?

File: tests/restApi/test_get_request.py (hunk 3 above)
Comment (ayan1229) [reply tests/restApi/test_get_request.py:17]: can remove. I put it for my own readability

File: tests/restApi/test_get_request.py (hunk 4 above)
Comment (ayan1229) [reply tests/restApi/test_get_request.py:37]: I haven't started UPDATE calls yet. Will be addressed in future PRs
For now it won't change as I have created this collection only for testing

File: CHANGELOG.md (hunk 1 above)
Comment (ayan1229) [reply CHANGELOG.md:1]: nope

File: tests/restApi/test_get_request.py
Comment (ayan1229) [reply tests/restApi/test_get_request.py:1]: I will change it to snake case

and change the test file to test_singel_collection_get_request.py 
If that's cool

File: CHANGELOG.md (hunk 1 above)
Comment (kylevillegas93) [reply CHANGELOG.md:1]: Ah ok all good. the extra newlines won't hurt anything.

File: tests/rest_api/test_get_collection.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    expected_title = "Works by Ayan"
+    actual_title = data.get("metadata", {}).get("title", "")
+   
+    assert actual_title == expected_title, f"Expected title '{expected_title}', but got '{actual_title}'"
+
+    first_publication_title = data.get('publications', [])[0].get('metadata', {}).get('title')
+    assert first_publication_title == "New Mexico magazine ", f"Expected first title 'New Mexico magazine ', but got {first_publication_title}"
+
+    second_publication_title = data.get('publications', [])[1].get('metadata', {}).get('title')
+    assert second_publication_title == "New Mexico magazine index. ", f"Expected second title 'New Mexico magazine index. ', but got {second_publication_title}"
```
Comment (kylevillegas93) [inline tests/rest_api/test_get_collection.py:31]: Recommend removing these assertions. We are less concerned about the data and more concerned that the API is returning 200 and a non-empty response.

File: tests/rest_api/test_get_collection.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    response = requests.get(url)
+    assert response.status_code == expected_status_code, (
+        f"API call failed.\n"
+        f"Expected status code: {expected_status_code}\n"
+        f"Actual status code: {response.status_code}\n"
+        f"URL: {url}\n"
+        f"Response text: {response.text[:100]}..."
+    )
+
+    data = response.json()
```
Comment (kylevillegas93) [inline tests/rest_api/test_get_collection.py:20]: Let's add an assertion to check this is not null

File: tests/rest_api/test_get_collection.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    url = BASE_URL + "/collection/3650664c-c8be-4d07-8d64-2d7003b02048"
+    expected_status_code = 200
+    response = requests.get(url)
+    assert response.status_code == expected_status_code, (
+        f"API call failed.\n"
+        f"Expected status code: {expected_status_code}\n"
+        f"Actual status code: {response.status_code}\n"
+        f"URL: {url}\n"
+        f"Response text: {response.text[:100]}..."
+    )
```
Comment (kylevillegas93) [inline tests/rest_api/test_get_collection.py:18]: Recommend using newlines for readability:

```
url = BASE_URL + "/collection/3650664c-c8be-4d07-8d64-2d7003b02048"

response = requests.get(url)
    
assert response.status_code == expected_status_code, (
        f"API call failed.\n"
        f"Expected status code: {expected_status_code}\n"
        f"Actual status code: {response.status_code}\n"
        f"URL: {url}\n"
        f"Response text: {response.text[:100]}..."
)
```

You don't necessarily need the newline on line 8.

File: tests/rest_api/test_get_collection.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import requests
+import pytest
+
+BASE_URL = "https://drb-api-qa.nypl.org"
+
+def test_get_collection():
+    url = BASE_URL + "/collection/3650664c-c8be-4d07-8d64-2d7003b02048"
+    expected_status_code = 200
+    response = requests.get(url)
+    assert expected_status_code is not None
```
Comment (kylevillegas93) [inline tests/rest_api/test_get_collection.py:10]: Should we be asserting that expected_status_code is not None? this is declared above.

File: tests/rest_api/test_get_collection.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    assert response.status_code == expected_status_code, (
```
Comment (kylevillegas93) [inline tests/rest_api/test_get_collection.py:11]: You can just do `assert response.status_code == 200, ...` as well

File: tests/rest_api/test_get_collection.py (hunk 10 above)
Comment (ayan1229) [reply tests/rest_api/test_get_collection.py:10]: yeah my bad on this.

**Review by kylevillegas93 — APPROVED** [review]

**Pull request merged**